File: Showdown/src/Card.php
<?php

namespace Showdown;

use Showdown\Enums\RankEnum;
use Showdown\Enums\SuitEnum;

class Card
{
    private RankEnum $rank;

    private SuitEnum $suit;

    public function __construct(RankEnum $rank, SuitEnum $suit)
    {
        $this->rank = $rank;
        $this->suit = $suit;
    }

    public function getRank(): RankEnum
    {
        return $this->rank;
    }

    public function getSuit(): SuitEnum
    {
        return $this->suit;
    }
}

File: showdown-go/entity/card.go
package entity

import (
	"showdown-go/enum"
)

// Card represents a playing card
type Card struct {
	rank enum.Rank
	suit enum.Suit
}

// GetRank returns the rank of the card
func (c *Card) GetRank() enum.Rank {
	return c.rank
}

// GetSuit returns the suit of the card
func (c *Card) GetSuit() enum.Suit {
	return c.suit
}

// NewCard creates a new card
func NewCard(rank enum.Rank, suit enum.Suit) *Card {
	return &Card{
		rank: rank,
		suit: suit,
	}
}
